src/tests/ftest/deployment/critical_integration.py

```diff
@@ -75,16 +76,33 @@ def test_passwdlessssh_versioncheck(self):
         result_dmg = dmg_version_list.count(dmg_version_list[0]) == len(dmg_version_list)
         result_client_server = daos_server_version_list[0][1:] == dmg_version_list[0]
 
-        if (result_daos_server and result_dmg and result_client_server):
+        # libfabric version check
+        all_nodes = self.hostlist_servers + self.hostlist_clients
+        all_nodes = list(dict.fromkeys(all_nodes))
```

Comment:
To union two `NodeSet` (and remove duplicate nodes)
```suggestion
        all_nodes = self.hostlist_servers | self.hostlist_clients
```

Reply:
Done